### ISSUE TITLE
OpenStack cloud provider is not deployed due to container name mismatch

### ISSUE THREAD

**Issue by ederst** [kubermatic/kubermatic#6748]
**What happened**:

After creating a user cluster I am getting the following error(s):

```
failed  to ensure Deployment  cluster-znftrnqpbz/openstack-cloud-controller-manager: failed to  generate object: failed to add apiserver.IsRunningWrapper: pod has no  container named "cloud-controller-manager"

# and IMHO this is a follow up error b/c the cloud controller does not get deployed:

failed  to create initial MachineDeployment: Internal error occurred: failed  calling webhook "machine-controller.kubermatic.io-machinedeployments":  Post  "https://machine-controller-webhook.cluster-znftrnqpbz.svc.cluster.local./machinedeployments?timeout=10s":  Service Unavailable
```

This means that kubermatic cannot create the openstack cloud controller because it cannot find the name of the container name specified here:

https://github.com/kubermatic/kubermatic/blob/503de88224894a1932a05891ff5bac09b8e9885f/pkg/resources/cloudcontroller/deployment.go#L61-L66

**What is the expected behavior**:

OpenStack cloud controller manager should be successfully deployed.

**How to reproduce the issue**:

I think this is reproducible by using version 2.16.x and the external OpenStack cloud controller.

**Anything else we need to know?**

I think the issue is that `cloudcontroller/deployment.go` uses the variable here for the container name:

https://github.com/kubermatic/kubermatic/blob/503de88224894a1932a05891ff5bac09b8e9885f/pkg/resources/cloudcontroller/deployment.go#L33

However when the resource is created in `cloudcontroller/openstack.go`, it uses this variable:

https://github.com/kubermatic/kubermatic/blob/503de88224894a1932a05891ff5bac09b8e9885f/pkg/resources/cloudcontroller/openstack.go#L34

https://github.com/kubermatic/kubermatic/blob/503de88224894a1932a05891ff5bac09b8e9885f/pkg/resources/cloudcontroller/openstack.go#L101-L114

Funnily enough this is not the case in the master, where `ccmContainerName` is used instead of `osName`:

https://github.com/kubermatic/kubermatic/blob/16260d4a74785f68f492d0a77c5cc48fb95650be/pkg/resources/cloudcontroller/openstack.go#L101-L114

Looking at the blame view, the lines got edited by the same PR (https://github.com/kubermatic/kubermatic/pull/6588), so this might be some merge issue?

**Information about the environment**:
Kubermatic version: 2.16.5

**Comment by ederst** [comment]
So the bug was probably introduced with the backport of the Hetzner cloud support: https://github.com/kubermatic/kubermatic/pull/6619

In `cloudcontroller/deployment.go` the variable `ccmContainerName` was introduced:

<img width="1199" alt="image" src="https://user-images.githubusercontent.com/2979051/112162485-64865500-8bec-11eb-9c49-2d446dc2e600.png">

<img width="634" alt="image" src="https://user-images.githubusercontent.com/2979051/112162552-7536cb00-8bec-11eb-8246-597efb64855c.png">

But unfortunately it stayed the same (`osName = "openstack-cloud-controller-manager"`) in `cloudcontroller/openstack`:

<img width="1269" alt="image" src="https://user-images.githubusercontent.com/2979051/112162651-88499b00-8bec-11eb-8180-1d3fae73dbc9.png">

**Comment by irozzo-1A** [comment]
Hi @ederst, thanks for reporting this and for the detailed explanation. I agree with your analysis, this was introduced in #6619. I'll double-check why this was not detected by our tests.

**Comment by ederst** [comment]
@irozzo-1A NP, was trying to fix it myself too (https://github.com/ederst/kubermatic/commit/95d6ef4a5da13f2bb43c253c12ce3c3d72ffa3c6). Wanted to test it first but have not had time, you guys are too fast. ;)

**Comment by irozzo-1A** [comment]
Oh sorry @ederst, I had to ask you first. Next time don't hesitate to mention that you are willing to work on the fix ;-)

**Comment by ederst** [comment]
@irozzo-1A really no problem, was not that time consuming of a fix, and was not sure on how you guys handle the backport, but looks like it was just a simple pr; so next time maybe ;)

**Comment by ederst** [comment]
@irozzo-1A ic the fix is already released in v2.16.7, i will try to test it asap.

> I'll double-check why this was not detected by our tests.

have you found out by any chance why the tests were not detecting this?

**Comment by irozzo-1A** [comment]
Thx @ederst let me know if you still find issues.
At the moment I improved a unit test that is covering resources generation to prevent this from happening again:
https://github.com/kubermatic/kubermatic/pull/6750/commits/6eef8e203d3a2b788fc2f6e00b01411d8191ea14
I still need to double-check why the conformance tests did not detect it.

**Comment by irozzo-1A** [comment]
the issue was solved by #6750
